src/Xeno/Types.hs

```diff
@@ -7,12 +7,13 @@ module Xeno.Types where
 
 import Control.DeepSeq
 import Control.Exception
+import Data.ByteString (ByteString)
 import Data.Typeable
 import GHC.Generics
 
 data XenoException
   = XenoStringIndexProblem
-  | XenoParseError
+  | XenoParseError ByteString
```

Comment:
This probably causes a speed hit on exception throwing. I think I had a message originally.